crates/cxx-qt-gen/src/naming/type_names.rs

```diff
@@ -139,14 +142,25 @@ impl TypeNames {
         Ok(())
     }
 
+    fn populate_qobject(&mut self, qobject: &ParsedQObject) -> Result<()> {
```

Comment:
this makes the naming dependent on the parser which is slightly unfortunate for making it reusable later. Maybe later we could have an intermediate struct this is populated from ?

Reply:
Right, that's true, but also already the case with the previous setup.

When we move this module out into its own crate, these functions will need to remain inside cxx-qt-gen, potentially as a free function.
They are the "naming" phase, so need to know how to extract the type names out of the parser data.